### ISSUE TITLE
GIT has been added to 🤗 Transformers

### ISSUE THREAD

**Issue by NielsRogge** [microsoft/GenerativeImage2Text#34]
Hi folks!

Impressive work :) as I really liked this work I decided to contribute it to 🤗 Transformers.

Documentation can be found here: https://huggingface.co/docs/transformers/main/en/model_doc/git

Users can now use GIT in a few lines of code :)

Here's a demo notebook illustrating inference with GIT for image/video captioning, and image/video QA: https://github.com/NielsRogge/Transformers-Tutorials/blob/master/GIT/Inference_with_GIT_for_image_video_captioning_and_image_video_QA.ipynb.

**Comment by FBehrad** [comment]
Does this model work with the GIT_LARGE model too?

**Comment by NielsRogge** [comment]
Yes, all GIT-large checkpoints can also be found on the hub, see: https://huggingface.co/models?other=git

**Comment by Neleac** [comment]
What's the best way to detach the text decoder from the image encoder? I want to experiment with other encoders than ViT, and finetune with the GIT text decoder. I've tried using [VisionEncoderDecoderModel](https://huggingface.co/docs/transformers/main/en/model_doc/vision-encoder-decoder#vision-encoder-decoder-models) but can't remove the GitVisionModel. Does Hugging Face have tools for what I want to do or should I write a custom class instead?

**Comment by devilismyfriend** [comment]
is there a chance to get a training examples script using transformers? thanks for your work!

**Comment by amsword** [comment]
Thanks a lot for integrating it to transformers @NielsRogge . Just now, I added three other checkpoints, which is re-trained by removing some offensive captions in cc12m dataset. can you help integrate these pre-trained checkpoint also?

![image](https://user-images.githubusercontent.com/3356680/211131642-e50b8347-720b-4a47-8fdc-9cd36ecbf17b.png)

**Comment by amsword** [comment]
> 

I'm not sure how to use transformers for this task, but if you use this repo, you can pay attention to the following function

https://github.com/microsoft/GenerativeImage2Text/blob/e1d90a2425e6f4c3e83e6689862b955e0029cbcc/generativeimage2text/model.py#L55

**Comment by amsword** [comment]
> What's the best way to detach the text decoder from the image encoder? I want to experiment with other encoders than ViT, and finetune with the GIT text decoder. I've tried using [VisionEncoderDecoderModel](https://huggingface.co/docs/transformers/main/en/model_doc/vision-encoder-decoder#vision-encoder-decoder-models) but can't remove the GitVisionModel. Does Hugging Face have tools for what I want to do or should I write a custom class instead?

@Neleac 
I'm not sure how to use transformers for this task, but if you use this repo, you can pay attention to the following function

https://github.com/microsoft/GenerativeImage2Text/blob/main/generativeimage2text/model.py#L55

**Comment by rumusan** [comment]
Thank @amsword and @NielsRogge for the great work. I find that the huggingface-version model can be faster, what is the main reason?

**Comment by NielsRogge** [comment]
Could you clarify "can be faster"?

**Comment by artificialguybr** [comment]
> Thanks a lot for integrating it to transformers @NielsRogge . Just now, I added three other checkpoints, which is re-trained by removing some offensive captions in cc12m dataset. can you help integrate these pre-trained checkpoint also?
> 
> ![image](https://user-images.githubusercontent.com/3356680/211131642-e50b8347-720b-4a47-8fdc-9cd36ecbf17b.png)

Yes. Adding those new checkpoints would be quite interesting. Would @NielsRogge be able to do this?

**Comment by NielsRogge** [comment]
I've uploaded those checkpoints! Find them here:

* [GIT-large-r](https://huggingface.co/microsoft/git-large-r)
* [GIT-large-r-coco](https://huggingface.co/microsoft/git-large-r-coco)
* [GIT-large-r-textcaps](https://huggingface.co/microsoft/git-large-r-textcaps)

@amsword I guess there are no plans to open-source the model referred to as "GIT" in the paper itself?

**Comment by amsword** [comment]
@NielsRogge , right. currently, only base and large.

**Comment by altryne** [comment]
Thanks for this work @NielsRogge !!

**Comment by tiesanguaixia** [comment]
> What's the best way to detach the text decoder from the image encoder? I want to experiment with other encoders than ViT, and finetune with the GIT text decoder. I've tried using [VisionEncoderDecoderModel](https://huggingface.co/docs/transformers/main/en/model_doc/vision-encoder-decoder#vision-encoder-decoder-models) but can't remove the GitVisionModel. Does Hugging Face have tools for what I want to do or should I write a custom class instead?

Hi! I have a similar plan, I see the code in _model.py_ have the _get_image_encoder_ and _get_git_model_ function independently, I guess if I can use them to seperate the text decoder from the image encoder, or did you resolve this problem? Could you please tell me how to accomplish this? Thanks a lot!
